File: nodejs/src/questionnaire/inputMod.js
module.exports.getSelection = async function (input) {
    console.log("==================================================")
    let selection

    try {
        selection = await input("Enter a selection (Q to quit): ")
        selection = selection.toUpperCase()

    } catch (e) {
        console.log(e)
        selection = ""
    } finally {
        return selection
    }
}

File: src/lib/r/src/questionnaire/input_mod.r
####################################################
questionnaireInput <- function(prompt) {
    selection <- ""

    tryCatch(
        {
         cat(prompt)  
         selection <- readLines("stdin",n=1)  
        },
        error=function(err) {
            selection <- ""
        },
        finally=function() {
            return(selection)
        }
    )
}
####################################################

questionnaireGetSelection <- function() {
    cat("==================================", "\n")
    selection <- questionnaireInput("Enter your choice (Q to quit): ")
    selection <- toupper(selection)
    return(selection)
}
